platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -1150,6 +1208,24 @@ bool OfflineDatabase::evict(uint64_t neededFreeSize) {
     return true;
 }
 
+std::exception_ptr OfflineDatabase::setMaximumAmbientCacheSize(uint64_t size) try {
+    maximumAmbientCacheSize = size;
```

Comment:
If db throws, maximumAmbientCacheSize will be set to size, I think it would be better to set internal maximumAmbientCacheSize only when db operations are successful

Reply:
Nice catch